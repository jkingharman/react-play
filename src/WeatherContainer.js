import React, { Component } from 'react';
import WeatherImage from './WeatherImage';
import LocationInput from './LocationInput';
import LocationBtn from './LocationBtn';

class WeatherContainer extends React.Component {

	constructor(props) {
  	super(props);
  	this.state = {currentWeather: "default", currentLocation: "London"};
    this.handleChange = this.handleChange.bind(this)
    this.handleClick = this.handleClick.bind(this)
  }

	componentWillMount() {
  	this.getWeather()
  }

  handleClick() {
   this.getWeather()
  }

  handleChange(e) {
  	this.setState({currentLocation: e.target.value})
  }

  parseResponse(res) {
  	return {
    	weather: res.weather[0].icon,
      temp: res.main.temp
    }
  }


	getWeather() {
   // refactor promise structure when writing for real request
  	let response = new Promise((resolve) => {
    let url = "https://api.openweathermap.org/data/2.5/weather?q=" + this.state.currentLocation + "&APPID=d67b6ea31078ccea4bd77846d9569c02"

    resolve(fetch(url))
    });

    response.then(res => {
  		res.json()
      	.then(json => {
          if (json.cod == "200") {
            this.setState({currentWeather: this.parseResponse(json)})
          } else if (json.cod == "404") {
            this.setState({currentWeather: {weather: "unknown", temp: "unknown"}})
          } else {
            console.log("error")
          }
        })
    })
  };

  render() {
   return (
   <div className="container">
   <WeatherImage weather={this.state.currentWeather}/>
   <LocationInput onChange={this.handleChange}/>
   <LocationBtn onClick={this.handleClick}/>
   </div>
   )
  }
}

export default WeatherContainer;
