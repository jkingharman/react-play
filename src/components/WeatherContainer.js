import React, { Component } from "react";
import WeatherImage from "./WeatherImage";
import LocationInput from "./LocationInput";
import LocationBtn from "./LocationBtn";

class WeatherContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentWeather: "",
      currentLocation: "London"
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  componentWillMount() {
    this.getWeather();
  }

  handleClick() {
    this.getWeather();
  }

  handleChange(e) {
    this.setState({
      currentLocation: e.target.value
    });
  }

  toCelcius(kelvin) {
    return (kelvin - 273).toFixed();
  }

  parseResponse(json) {
    if (json.cod === 200) {
      this.setState({
        currentWeather: {
          weather: json.weather[0].icon,
          temp: this.toCelcius(json.main.temp)
        }
      });
    } else if (json.cod === "404") {
      this.setState({
        currentWeather: {
          weather: "unknown",
          temp: "unknown"
        }
      });
    } else {
      console.log("error");
    }
  }

  getWeather() {
    fetch(
      `https://api.openweathermap.org/data/2.5/weather?q=${
        this.state.currentLocation
      }&APPID=d67b6ea31078ccea4bd77846d9569c02`
    ).then(res => {
      res.json().then(this.parseResponse.bind(this));
    });
  }

  render() {
    return (
      <div className="container">
        <WeatherImage weather={this.state.currentWeather} />
        <LocationInput
          location={this.state.currentLocation}
          onChange={this.handleChange}
        />
        <LocationBtn onClick={this.handleClick} />
      </div>
    );
  }
}

export default WeatherContainer;
