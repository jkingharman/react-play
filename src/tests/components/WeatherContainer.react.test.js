import React from 'react';
import renderer from 'react-test-renderer';
import WeatherContainer from '../../components/WeatherContainer.js';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <WeatherContainer/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
