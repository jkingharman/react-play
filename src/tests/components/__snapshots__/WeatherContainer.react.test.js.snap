// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="container"
>
  <div
    className="gallery-container"
  >
    <img
      alt="weather"
    />
    <h2>
       
       °C 
    </h2>
  </div>
  <form>
    <input
      type="text"
      value="London"
    />
  </form>
  <button
    onClick={[Function]}
  >
     
  </button>
</div>
`;
